semana 17/sg_tables_logic.py

```diff
@@ -0,0 +1,54 @@
+import PySimpleGUI as sg 
+from validators import Validators
+
+class CategoryDataManager:
+
+    def __init__(self, category_data_list=None):
+        self.category_data_list = category_data_list if category_data_list is not None else []
+
+    def process_user_category(self, category_window, values):
+        user_category_input = values["-CATEGORY-"]
+
+        if Validators.validate_user_input(user_category_input):
+            if Validators.is_duplicate_category(user_category_input, self.category_data_list):
```

Comment:
if Validators.validate_user_input(user_category_input) and Validators.is_duplicate_category(user_category_input, self.category_data_list):